crates/prql_compiler/src/semantic/ast_expand.rs

```diff
@@ -86,7 +64,47 @@ fn expand_expr_kind(value: ExprKind) -> pl::ExprKind {
         ),
         ExprKind::Param(v) => pl::ExprKind::Param(v),
         ExprKind::Internal(v) => pl::ExprKind::Internal(v),
+    };
+
+    pl::Expr {
+        kind,
+        span: expr.span,
+        alias: expr.alias,
+        id: None,
+        target_id: None,
+        target_ids: Vec::new(),
+        ty: None,
+        lineage: None,
+        needs_window: false,
+        flatten: false,
+    }
+}
+
+fn expand_exprs(exprs: Vec<prql_ast::expr::Expr>) -> Vec<pl::Expr> {
+    exprs.into_iter().map(expand_expr).collect::<Vec<_>>()
+}
+
+#[allow(clippy::boxed_local)]
+fn expand_expr_box(expr: Box<prql_ast::expr::Expr>) -> Box<pl::Expr> {
+    Box::new(expand_expr(*expr))
+}
+
+fn desugar_pipeline(mut pipeline: prql_ast::expr::Pipeline) -> pl::Expr {
```

Comment:
This function deserves some comment explaining what it does ... what the logic behind it is.

Reply:
Also the signature should probably be changed to return `pl::FuncCall` to make it clear what it gets desugared to.

Reply:
It cannot return `pl::FuncCall` because it needs to set the span and alias. I added a comment about it.